### PR TITLE
Adopt Stevens' wind axes convention

### LINKED ISSUE #106: Alpha and beta ranges in `OperatingPoint` may need tighter restriction
# Bug Description

`OperatingPoint` currently restricts `alpha` and `beta` to the range (-180.0, 180.0], but this may still be too permissive. Large absolute values within this range may cause silent numerical failures in the solvers.

**Location(s):** `pterasoftware/operating_point.py`

## Expected Behavior

If testing reveals that extreme `alpha` or `beta` values (within the current (-180.0, 180.0] range) break the solvers, the ranges should be tightened with a clear error or warning.

## Steps to Reproduce

1. Create an `OperatingPoint` with `alpha` or `beta` set to extreme values within the current allowed range (e.g., 90, 179).
2. Run a steady solver and inspect the results for numerical issues.

## Additional Context

This issue is referenced by a `TODO` comment in `OperatingPoint`.

### PR DESCRIPTION
**Pull request by camUrban**
# Description

This pull request changes how Ptera Software constructs wind axes from body axes. Read as an active rotation that carries the body axes' basis vectors onto the wind axes' basis vectors, the construction now rotates about the body y axis through -alpha first, and then about the resulting z axis through beta, following Section 2.3 of "Aircraft Control and Simulation" by Stevens, Lewis, and Johnson (2016). This keeps the wind z axis in the body xz plane for every pair of angles, so lift is independent of sideslip, and it moves the poles of the parameterization from alpha = +/-90.0 to beta = +/-90.0. Beta is now restricted to [-90.0, 90.0], alpha keeps its (-180.0, 180.0] range, and alpha is defined to be 0.0 when |beta| = 90.0, so every velocity direction in body axes corresponds to exactly one pair of angles. `alpha_and_beta_from_vInf_BP1` is updated to be the exact inverse, including wrapping an alpha of -180.0 to 180.0 and returning alpha = 0.0 at the beta boundaries rather than letting signed zeros pick a value. Thanks to @CyberSarvesh, whose round-trip script on issue #106 put me on the path to this bug.

Three related changes ride along. The side force is replaced by a crosswind force equal to the negative of the wind axes' y force component, so all three named force components negate their wind axes components as in the cited source. The trim analyses now place the Airplane's weight along the direction of `g_E` and rotate it into each trial's wind axes, and each trial resolves its own attitude to level flight rather than copying the base OperatingPoint's attitude. The wind axes section of `docs/AXES_POINTS_AND_FRAMES.md` is rewritten for the new convention with provenance notes, and every claim it makes now has a unit test. The serialization format version and the convergence solve cache version are bumped, since the meaning of a saved or cached load with nonzero sideslip has changed. Results change only for simulations with alpha and beta both nonzero, and the side force names are removed rather than kept with a flipped sign so consumers break loudly.

## Motivation

The previous construction applied the same two rotations in the opposite order: read as an active rotation carrying the body axes' basis vectors onto the wind axes' basis vectors, it rotated about the body z axis through beta first and then about the resulting y axis through -alpha. With that order the wind z axis leaves the body xz plane whenever both angles are nonzero, so lift depended on sideslip, which does not match the description in `docs/AXES_POINTS_AND_FRAMES.md`. It also put the poles at alpha = +/-90.0, so a condition with alpha past 90.0 required a branch change in alpha, and the (-180.0, 180.0] range on both angles gave every velocity direction two names. The two conventions agree whenever alpha or beta is zero, which covers every existing example, validation script, and integration test, so the affected class of problems is small.

Issue #106 was opened on a concern about gimbal lock and a possible need to restrict alpha and beta. The forward transforms are orthonormal for every pair of angles, so that concern is not reproduced. While investigating it, @CyberSarvesh found that the extractor returns beta = -180.0 for exact backward flight, which `OperatingPoint` rejects, and that (180.0, 0.0) and (0.0, 180.0) give the same freestream. The -180.0 result comes from a negative zero component reaching the two-argument arctangent, and both findings are symptoms of the rotation order. The range restriction the issue asked about therefore lands on beta rather than alpha.

The trim analyses also placed the weight along the wind z axis, which is only correct when wind axes coincide with Earth axes. That is the attitude the default `angles_E_to_BP1_izyx` resolves to, so the existing examples and tests were unaffected, but a base OperatingPoint with an explicit attitude would have carried that attitude into every trial and tilted the flight path relative to Earth as alpha and beta varied. The side force sign is changed because Stevens writes the aerodynamic force in wind axes as (-D, -C, -L), negating all three components, while Ptera Software negated only the x and z components.

## Relevant Issues

Closes #106. The gimbal lock concern that issue was opened on is not reproduced, but the round-trip findings in its discussion led to the rotation order bug this pull request fixes, and the fix answers the issue's range question.

## Changes

* `OperatingPoint.T_pas_BP1_CgP1_to_W_CgP1` in `operating_point.py` now calls `generate_rot_T` with `order="zyx"` instead of `order="xyz"` for the same `(0.0, -alpha, beta)` angles, and the default `angles_E_to_BP1_izyx` resolution uses the same sequence so wind axes still coincide with Earth axes by default.
* `OperatingPoint.__init__` validates beta in [-90.0, 90.0] instead of (-180.0, 180.0], and raises a `ValueError` if |beta| = 90.0 and alpha is not 0.0. The `alpha`, `beta`, and `g_E` docstrings are updated to match.
* `alpha_and_beta_from_vInf_BP1` in `_transformations.py` now recovers beta with an arcsine of the body y component and alpha with a two-argument arctangent of the body z and x components, wraps an alpha of exactly -180.0 to 180.0, and returns alpha = 0.0 when |beta| = 90.0 or when both arctangent inputs are zero.
* The side force is replaced by a crosswind force equal to `-fY_W`. `sideForce_W` and `sideForceCoefficient_W` on `Airplane` and `sideForce_W` on `Panel` become `crosswindForce_W` and `crosswindForceCoefficient_W`. `finalSideForces_W`, `finalSideForceCoefficients_W`, `finalMeanSideForces_W`, and `finalMeanSideForceCoefficients_W` on `CoreUnsteadyProblem` become `finalCrosswindForces_W`, `finalCrosswindForceCoefficients_W`, `finalMeanCrosswindForces_W`, and `finalMeanCrosswindForceCoefficients_W`. The `"side force"` scalar type for `draw` and `animate` becomes `"crosswind force"`, and the plot and log labels in `output.py` change accordingly.
* `analyze_steady_trim` and `analyze_unsteady_trim` in `trim.py` build the weight force in Earth axes as the Airplane's weight times the unit vector of `g_E`, rotate it into each trial's wind axes with `T_pas_E_CgP1_to_W_CgP1`, and add `externalFX_W` along the wind x axis. Both raise a `ValueError` if `g_E` is zero.
* Both trim functions leave each trial's `angles_E_to_BP1_izyx` unset so it resolves to level flight at the trial's own alpha and beta, and both raise a `ValueError` if the base OperatingPoint's attitude does not already resolve to level flight, since the trials would otherwise silently discard it.
* `examples/analyze_steady_trim.py` and `examples/analyze_unsteady_trim.py` set a nonzero `g_E` on their OperatingPoints, with a comment explaining that only its direction matters.
* `_FORMAT_VERSION` in `_serialization.py` is bumped from 28 to 29 and `_SOLVE_CACHE_VERSION` in `_convergence_cache.py` from 7 to 8, so files and caches written under the old wind axes meaning fail to load rather than being read under the new one. The `_FORMAT_VERSION` comment now says it also increments when the meaning of a serialized value changes.
* The wind axes section of `docs/AXES_POINTS_AND_FRAMES.md` is rewritten for the new rotation series, states the accepted ranges and the |beta| = 90.0 exception, defines beta as the angle from the body xz plane to the velocity and alpha as the angle from the body x axis to the velocity's projection onto that plane, and limits the sign-convention description to small |alpha| and |beta|. Provenance notes cite Stevens, Lewis, and Johnson (2016) for body axes, wind axes, and the two angles, and Drela (2014) for geometry axes. The crosswind force replaces the side force in the named load components, and `docs/CLASSES_AND_IMMUTABILITY.md` reflects the renamed properties.
* `tests/unit/test_operating_point.py` gains a test for every claim in the rewritten wind axes section. The claims that hold for every pair of angles are swept over the full accepted ranges in five degree increments at the default attitude, including a golden matrix written out from the source independently of `generate_rot_T`, the alignment of the wind x axis with the velocity, the wind z axis lying in the body xz plane, the extractor round trip, the uniqueness of each pair, and the rejection of every nonzero alpha at |beta| = 90.0. The small-angle sign conventions and the attitude independence of the construction are swept over a grid of attitudes as well.
* `tests/unit/test_transformations.py` updates the extractor's round trip test for the new convention and adds tests that exact backward flight yields alpha = 180.0 and that a freestream along the body y axis yields beta = +/-90.0 with alpha = 0.0 regardless of signed zeros. The OperatingPoint fixtures with beta at or near its boundaries move from 180.0 to 90.0.
* `tests/unit/test_unsteady_trim.py` and `tests/integration/test_steady_trim.py` add tests that a zero `g_E` and a non-level base attitude are rejected, and the unit tests add a stubbed-solver test that each trial resolves to level flight and balances the weight placed along a tilted `g_E`. Existing trim tests set a nonzero `g_E`.
* The expected outputs under `docs/examples_expected_output/` and the `getting_started.ipynb` tutorial are regenerated, and the remaining tests are updated for the crosswind force names.
* Both trim analyses now reject alpha bounds outside (-180.0, 180.0] and beta bounds outside (-90.0, 90.0) up front. The search builds an OperatingPoint from every trial and varies alpha and beta independently, so bounds outside OperatingPoint's ranges, or beta bounds touching the poles where only alpha = 0.0 is accepted, would otherwise let a trial raise partway through the search.

## Dependency Updates

None.

## Change Magnitude

**Major**: Large change that adds significant new functionality, changes existing behavior, or may affect many parts of the codebase.

## Checklist (check each item when completed or not applicable)

* [x] I am familiar with the current [contribution guidelines](https://github.com/camUrban/PteraSoftware/blob/main/CONTRIBUTING.md).
* [x] PR description links all relevant issues and follows this template.
* [x] My branch is based on `main` and is up to date with the upstream `main` branch.
* [x] All calculations use S.I. units.
* [x] Code is formatted with [black](https://github.com/psf/black) (line length = 88).
* [x] Code is well documented with block comments where appropriate.
* [x] Any external code, algorithms, or equations used have been cited in comments or docstrings.
* [x] All new modules, classes, functions, and methods have docstrings in [reStructuredText format](https://realpython.com/documenting-python-code/), and are formatted using [docformatter](https://github.com/PyCQA/docformatter) (`--in-place --black`). See the [style guide for type hints and docstrings](https://github.com/camUrban/PteraSoftware/blob/main/docs/TYPE_HINT_AND_DOCSTRING_STYLE.md) for more details.
* [x] All new classes, functions, and methods in the `pterasoftware` package use type hints. See the [style guide for type hints and docstrings](https://github.com/camUrban/PteraSoftware/blob/main/docs/TYPE_HINT_AND_DOCSTRING_STYLE.md) for more details.
* [x] If any major functionality was added or significantly changed, I have added or updated tests in the `tests` package.
* [x] Code locally passes all tests in the `tests` package.
* [x] This PR passes the ReadTheDocs build check (this runs automatically with the other workflows).
* [x] This PR passes the `ascii-only`, `pre-commit-hooks`, and `zizmor` GitHub actions.
* [x] This PR passes the `lint` job of the `CI` GitHub action.
* [x] This PR passes the `test` jobs of the `CI` GitHub action.

Assisted-by: Claude Fable 5.1

### CODE REVIEW

File: pterasoftware/operating_point.py [hunk 1]
```diff
@@ -200,25 +206,38 @@ def __init__(
...
             alpha, "alpha", -180.0, False, 180.0, True
         )
         self._beta = _parameter_validation.number_in_range_return_float(
-            beta, "beta", -180.0, False, 180.0, True
+            beta, "beta", -90.0, True, 90.0, True
         )
+        if abs(self._beta) == 90.0 and self._alpha != 0.0:
+            raise ValueError(
+                "alpha must be 0.0 when the absolute value of beta is 90.0."
+            )
```
[the remark this answers is not in the document]
Comment (camUrban) [reply pterasoftware/operating_point.py:228]: This is a good point. I'll commit a fix.
